File: client/src/main/java/io/theam/client/commands/GetCustomerImageCommand.java
package io.theam.client.commands;

import com.github.rvesse.airline.annotations.Command;
import com.github.rvesse.airline.annotations.Option;
import io.theam.client.service.RestClient;
import io.theam.model.api.ImageData;
import io.theam.util.UtilBase64Image;
import org.apache.commons.lang3.StringUtils;

import java.awt.*;
import java.io.File;
import java.io.IOException;

@Command(name = "get", description = "Gets/downloads the image of a customer")
public class GetCustomerImageCommand extends BaseCommandWithId {

    @Option(name = "--show", description = "Opens default application to show the image")
    public boolean showImage = false;

    @Option(name = {"--file", "-f"}, description = "The path to store the downloaded image")
    public String imagePath;

    @Override
    protected void doRun() {

        ImageData image = new RestClient(username, password).getCustomerImage(customerId);

        if(StringUtils.isEmpty(imagePath)) {
            System.out.println("Image downloaded but not stored locally (no image path defined)");
            System.out.println(image.toString());
        }else{
            UtilBase64Image.INSTANCE.decoder(image.getFileData(), imagePath);

            if(showImage) {
                try {
                    Desktop.getDesktop().open(new File(imagePath));
                } catch (IOException e) {
                    throw new RuntimeException(e);
                }
            }
        }
    }
}

File: client/src/main/kotlin/io/theam/client/commands/customers/GetCustomerImageCommand.kt
package io.theam.client.commands.customers

import com.github.rvesse.airline.annotations.Command
import com.github.rvesse.airline.annotations.Option
import io.theam.client.commands.BaseCommandWithId
import io.theam.client.service.bodyOf
import io.theam.client.service.getEntity
import io.theam.client.service.withUrl
import io.theam.model.api.ImageData
import io.theam.util.decodeToFile
import java.awt.Desktop
import java.io.File

@Command(name = "get", description = "Gets/downloads the image of a customer")
class GetCustomerImageCommand : BaseCommandWithId() {

    @Option(name = ["--show"], description = "Opens default application to show the image")
    var showImage = false

    @Option(name = ["--file", "-f"], description = "The path to store the downloaded image")
    var imagePath: String? = null

    private fun getCustomerImage(): ImageData =
            bodyOf(restClient withUrl
                    "$host/customers/${customerId!!}/image" getEntity ImageData::class.java)

    override fun doRun() =
            getCustomerImage().let { image ->
               if ((imagePath ?: "").isBlank()) {
                    println("Image downloaded but not stored locally (no image path defined)")
                    println(image.toString())
                } else {
                    image.fileData decodeToFile imagePath!!
                    if (showImage) Desktop.getDesktop().open(File(imagePath!!))
                }
            }
}
